File: 프로그래머스/3/42628. 이중우선순위큐/이중우선순위큐.java
import java.util.*;

class Solution {
    public int[] solution(String[] operations) {
        PriorityQueue<Integer> maxQue = new PriorityQueue<>(Collections.reverseOrder());
        PriorityQueue<Integer> minQue = new PriorityQueue<>();

        for (String operation : operations) {
            String[] o = operation.split(" ");

            if (o[0].equals("I")) {
                maxQue.offer(Integer.parseInt(o[1]));
                minQue.offer(Integer.parseInt(o[1]));
            } else {
                if (o[1].equals("1")) {
                    minQue.remove(maxQue.poll());
                } else {
                    maxQue.remove(minQue.poll());
                }
            }
        }

        if (!maxQue.isEmpty()) {
            return new int[]{maxQue.poll(), minQue.poll()};
        } else {
            return new int[2];
        }
    }
}

File: 프로그래머스/lv3/42628. 이중우선순위큐/이중우선순위큐.kt
import java.util.*

class Solution {
    fun solution(operations: Array<String>): IntArray {
        val minQue = PriorityQueue<Int>()
        val maxQue = PriorityQueue<Int>(Collections.reverseOrder())
        
        operations.map { it.split(" ") }.forEach {
            if (it[0] == "I") {
                minQue.add(it[1].toInt())
                maxQue.add(it[1].toInt())
                return@forEach
            }
            
            if (minQue.isNotEmpty()) {
                if (it[1] == "1") {
                    minQue.remove(maxQue.poll())
                } else {
                    maxQue.remove(minQue.poll())
                }
            }
        }
        
        return if (minQue.isNotEmpty()) {
            intArrayOf(maxQue.poll(), minQue.poll())
        } else {
            IntArray(2)
        }
    }
}